File: app/src/main/java/fr/insapp/insapp/CustomTabsConnection.java
package fr.insapp.insapp;

import android.content.ComponentName;
import android.support.customtabs.CustomTabsClient;
import android.support.customtabs.CustomTabsServiceConnection;
import android.support.customtabs.CustomTabsSession;

/**
 * Created by thomas on 09/02/2017.
 */

public class CustomTabsConnection extends CustomTabsServiceConnection {

    private CustomTabsClient customTabsClient;
    private CustomTabsSession customTabsSession;

    @Override
    public void onCustomTabsServiceConnected(ComponentName name, CustomTabsClient client) {
        this.customTabsClient = client;
        customTabsClient.warmup(0L);
        this.customTabsSession = customTabsClient.newSession(null);
    }

    @Override
    public void onServiceDisconnected(ComponentName name) {

    }

    public CustomTabsSession getCustomTabsSession() {
        return customTabsSession;
    }
}

File: app/src/main/java/fr/insapp/insapp/components/CustomTabsConnection.kt
package fr.insapp.insapp.components

import android.content.ComponentName
import androidx.browser.customtabs.CustomTabsClient
import androidx.browser.customtabs.CustomTabsServiceConnection
import androidx.browser.customtabs.CustomTabsSession

/**
 * Created by thomas on 09/02/2017.
 */

class CustomTabsConnection : CustomTabsServiceConnection() {

    private var customTabsClient: CustomTabsClient? = null

    var customTabsSession: CustomTabsSession? = null
        private set

    override fun onCustomTabsServiceConnected(name: ComponentName, client: CustomTabsClient) {
        this.customTabsClient = client

        customTabsClient?.warmup(0L)
        this.customTabsSession = customTabsClient?.newSession(null)
    }

    override fun onServiceDisconnected(name: ComponentName) {

    }
}
